### PR TITLE
fix 503 error from opencollective

### PR DESCRIPTION
**Pull request by chenxsan**
closes https://github.com/webpack/webpack.js.org/issues/4211

### CODE REVIEW

File: src/utilities/fetch-supporters.js [hunk 1]
```diff
@@ -9,9 +9,24 @@ const asyncWriteFile = promisify(fs.writeFile);
...
 
-const graphqlEndpoint = 'https://api.opencollective.com/graphql/v2';
+let graphqlEndpoint =
+  'https://api.opencollective.com/graphql/v2';
+
+// https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v2/query/TransactionsQuery.ts#L81
+const graphqlPageSize = 1000;
+
+if (process.env && process.env.CI && process.env.OPENCOLLECTIVE_API_KEY) {
+  // use api key when deploying to production
```
Comment (chenxsan) [inline src/utilities/fetch-supporters.js:27]: Currently I'm using a api key from my own opencollective account, you can replace it here https://github.com/webpack/webpack.js.org/settings/secrets/actions

File: src/utilities/fetch-supporters.js [hunk 2]
```diff
@@ -32,10 +47,12 @@ const membersGraphqlQuery = `query account($limit: Int, $offset: Int) {
   }
 }`;
 
-const transactionsGraphqlQuery = `query account($limit: Int, $offset: Int) {
-  account(slug: "webpack") {
-    transactions(limit: $limit, offset: $offset, includeIncognitoTransactions: false) {
-      nodes {
+// only query transactions in last year
+const transactionsGraphqlQuery = `query transactions($dateFrom: ISODateTime, $limit: Int, $offset: Int) {
```
Comment (chenxsan) [inline src/utilities/fetch-supporters.js:51]: Now we query only data in last year.

File: src/utilities/fetch-supporters.js (hunk 1 above)
Comment (sokra) [reply src/utilities/fetch-supporters.js:27]: Do we really need an API key for that. That should be public data and it worked without before too.

Comment (chenxsan) [reply src/utilities/fetch-supporters.js:27]: ok, i'll remove it and let's see how the improved version works first.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -42,6 +42,7 @@ jobs:
         uses: JamesIves/github-pages-deploy-action@3.7.1
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OPENCOLLECTIVE_API_KEY: ${{ secrets.OPENCOLLECTIVE_API_KEY }}
```
Comment (montogeek) [inline .github/workflows/deploy.yml:45]: Which account holds this key?

Comment (chenxsan) [reply .github/workflows/deploy.yml:45]: No longer used, could be removed. PR here https://github.com/webpack/webpack.js.org/pull/4390

**Pull request merged**